src/reason_toolchain.ml

```diff
@@ -487,9 +510,14 @@ module JS_syntax = struct
        loop_handle_yacc supplier in_error checkpoint
     | I.HandlingError env ->
        if !Reason_config.recoverable then
+         (
+         let loc = last_token_loc supplier in
+         let state = state checkpoint in
```

Comment:
This has been a while so my memory may be blended -- I remember that Menhir functions are mostly side-effects free ? (e.g., checkpoint won't change after this function). I can take a deeper look to validate.